library/manageiq_alert.py

```diff
@@ -0,0 +1,284 @@
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+  description:
+    description:
+      - the alert definition description in manageiq. this is the primary key
+        used to match alert definitions, therefor always required
+    required: true
+    default: null
+  entity:
+    description:
+      - the entity to base the alert on in manageiq
+    required: false
+    choices: ['container_node', 'vm', 'host', 'storage', 'cluster', 'ems', 'miq_server', 'middleware_server']
+    default: null
+  expression:
+    description:
+      - the expression to evaluate
+    required: false
+    default: null
+  options:
+    description:
+      - Additional alert options, including the notification type and frequency
+    required: false
+    default: null
+  enabled:
+    description:
+      - whether the alert is active or not
+    required: false
+    default: true
+  state:
+    description:
+      - the state of the alert definition
+      - On present, it will create the alert if it does not exist, or update it
+        if the associated data is different
+      - On absent, it will delete the alert if it exists
+    required: false
+    choices: ['present', 'absent']
+    default: 'present'
+'''
+
+EXAMPLES = '''
+# Create a new alert in ManageIQ
+  manageiq_alert:
+    description: Test Alert 01
+    options:
+      notifications:
+        delay_next_evaluation: 60
+        evm_event: {}
+    entity: container_node
+    expression:
+      eval_method: dwh_generic
+      mode: internal
+    enabled: true
+    state: present
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+    miq_verify_ssl: False
+'''
+
+import os
+from manageiq_client.api import ManageIQClient as MiqApi
+
+
+class ManageIQAlert(object):
+    """ ManageIQ object to execute alert definitions management operations in manageiq
+
+    url            - manageiq environment url
+    user           - the username in manageiq
+    password       - the user password in manageiq
+    miq_verify_ssl - whether SSL certificates should be verified for HTTPS requests
+    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+
+    supported_entities = {
+        'container_node': 'ContainerNode', 'vm': 'Vm', 'miq_server': 'MiqServer', 'host': 'Host',
+        'storage': 'Storage', 'cluster': 'EmsCluster', 'ems': 'ExtManagementSystem',
+        'miq_server': 'MiqServer', 'middleware_server': 'MiddlewareServer'
+    }
+
+    def __init__(self, module, url, user, password, miq_verify_ssl, ca_bundle_path):
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password), verify_ssl=miq_verify_ssl, ca_bundle_path=ca_bundle_path)
+        self.changed       = False
```

Comment:
Is fine.  (1) ansible modules always run in separate process. (2) we do want ability to run multiple actions in tests, but both `main()` and tests always create new instance, run one action (create/update/etc), discard instance.